### ISSUE TITLE
Implement limit on Project.total_size

### ISSUE THREAD

**Issue by di** [pypa/warehouse#7446]
Now that we have the `Project.total_size` attribute, we should implement a limit on this value across all projects similar to per-file limit.

This would help catch projects that are a significant burden on our overall size on disk but remain under the 60MB per-file limit. It would also encourage projects that publish large nightlies to clean up older nightlies.

I think this limit should probably be something like 100GB, but happy to hear justification for different values. This should also be easy to adjust if need be.

**Comment by VikramJayanthi17** [comment]
I have submitted 2 PRs to implement this feature. I split up the internal and user-facing functionality so that enforcement can take place after admins have set custom limits for projects with abnormal sizes.

- [ ] Internal Changes, allowing the admin to set project size limits via the admin dashboard.(#8128)
- [ ] User-facing changes, enforcing the project size limit and information on how to request more/exemptions.(#8129)

What the default limit should be is under discussion in #8128. 

Please let me know if you all have any suggestions/changes, thanks.

**Comment by di** [comment]
There's an open question here about what the max default size should be. Running some statistics on the current project sizes:

```
warehouse=> select k, percentile_disc(k) within group (order by total_size) from projects, generate_series(0.99, 1, 0.001) as k group by k;
   k   | percentile_disc
-------+-----------------
  0.99 |       164417777
 0.991 |       190902738
 0.992 |       223755046
 0.993 |       274796535
 0.994 |       331700601
 0.995 |       410206567
 0.996 |       522704868
 0.997 |       709843942
 0.998 |      1118441107
 0.999 |      2569971350
 1.000 |    229525724418
(11 rows)
```

A limit of 2.5GB would cover 99.9% of projects. Given the current median project size:

```
warehouse=> select percentile_disc(0.5) within group (order by total_size) from projects;
 percentile_disc
-----------------
           41042
(1 row)
```

the average project would ~never hit this limit. Given the current per-file limit of 60MB, that would be ~40 60MB releases before a project would hit the limit. 

We'd need to grandfather in 241 packages currently larger than 2.5GB:

```
warehouse=> select count(*) from projects where total_size > 2569971350;
 count
-------
   241
(1 row)
```

Some additional details on the 99.99th percentile:

```
warehouse=> select k, percentile_disc(k) within group (order by total_size) from projects, generate_series(0.999, 1, 0.0001) as k group by k;
   k    | percentile_disc
--------+-----------------
  0.999 |      2569971350
 0.9991 |      2980650928
 0.9992 |      3510247758
 0.9993 |      4113658386
 0.9994 |      4938314126
 0.9995 |      6474534391
 0.9996 |      8012387229
 0.9997 |     11299904110
 0.9998 |     18084734831
 0.9999 |     77940281725
 1.0000 |    229525724418
(11 rows)
```

**Comment by di** [comment]
@dstufft says:

> I think that honestly the difference between 2.5 and say.. even 5 or 10GB is pretty small.

I'm fine with just making this 10GB, it would mean that we only need to grandfather in 73 projects instead.

**Comment by di** [comment]
I grandfathered in all existing projects with `Project.total_size` >= 10GB. I set their limits to roughly twice their current size, minus ~20%, rounded to the nearest 10GB. My thought is that PyPI's total size is roughly doubling every year, and that the rate of growth of these should probably fall under that curve.

I wouldn't expect any of the projects on https://pypi.org/stats to request _total size_ increases in the next ~1 year. I think we can give them _file size_ increases liberally though.
